File: sdk-ts/src/exceptions.ts
/** Mirrors sdk/eminidatabase_sdk/exceptions.py's ApiError exactly — the API's own
 * error shape (`{"detail": ...}`) surfaced as a typed exception rather than
 * forcing callers to inspect status codes themselves. */
export class ApiError extends Error {
  readonly statusCode: number;
  readonly detail: string;

  constructor(statusCode: number, detail: string) {
    super(`API error ${statusCode}: ${detail}`);
    this.name = "ApiError";
    this.statusCode = statusCode;
    this.detail = detail;
  }
}

File: sdk/eminidatabase_sdk/exceptions.py
class ApiError(Exception):
    """Raised for any non-2xx response — mirrors the API's own error shape
    (`{"detail": ...}`) rather than forcing callers to inspect status codes
    themselves."""

    def __init__(self, status_code: int, detail: str):
        super().__init__(f"API error {status_code}: {detail}")
        self.status_code = status_code
        self.detail = detail
